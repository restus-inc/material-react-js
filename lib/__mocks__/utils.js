/* global jest */
const utils = jest.genMockFromModule('../utils');

const newId = () => 'a0123456';
utils.newId = newId;

export default utils;
export { newId };
