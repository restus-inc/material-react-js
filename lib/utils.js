/**
 * Utilities for Material React.
 * @module material-react/lib/utils
 */

/**
 * Generates a rundom ID. The generated ID starts with basic latin child alphabet, and
 * subsequent characters consist of child alphanumeric character. (i.e. it matches
 * `/[a-z][a-z0-9]{digit-1}/`)
 * @param {number} [digit=8] The number of digit of the generated ID
 * @returns {string} The ID generated
 */
export function newId(digit = 8) {
  return (
    (Math.floor(Math.random() * 26) + 10) * (36 ** (digit - 1))
    + Math.floor(Math.random() * (36 ** (digit - 1)))
  ).toString(36);
}

/**
 * Converts the event name to event handler name.
 * @param {string} eventName The event name that is converted to the handler name
 * @return {string} The event handler name converted
 */
export function handlerName(eventName) {
  return `on${eventName.charAt(0).toUpperCase()}${eventName.slice(1)}`;
}
