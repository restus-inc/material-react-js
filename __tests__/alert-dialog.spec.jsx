/* global describe, it, expect */
import 'regenerator-runtime/runtime';
import React from 'react';
import { htmlOfRendering } from './utils';

import AlertDialog from '../lib/alert-dialog';

describe('AlertDialog component', () => {
  it('supports simple dialogs', () => {
    let buttons = [{ action: 'ok', label: 'OK', isDefault: true }];
    expect(htmlOfRendering(
      <AlertDialog content="foo" buttons={buttons} onClosed={() => {}}/>,
    )).resolves.toMatchSnapshot();

    buttons = [
      { action: 'cancel', label: 'Cancel', isDefault: true },
      { action: 'ok', label: 'OK' },
    ];
    expect(htmlOfRendering(
      <AlertDialog content="foo" className="bar" buttons={buttons} onClosing={() => {}}/>,
    )).resolves.toMatchSnapshot();
  });
});
