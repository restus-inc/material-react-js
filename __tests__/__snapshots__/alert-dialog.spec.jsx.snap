// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertDialog component supports simple dialogs 1`] = `
"<div
  class=\\"mdc-dialog\\"
>
  <div
    class=\\"mdc-dialog__container\\"
  >
    <div
      aria-modal=\\"true\\"
      class=\\"mdc-dialog__surface\\"
      role=\\"alertdialog\\"
    >
      <div
        class=\\"mdc-dialog__content\\"
      >
        foo
      </div>
      <div
        class=\\"mdc-dialog__actions\\"
      >
        <button
          class=\\"mdc-button mdc-dialog__button\\"
          data-mdc-dialog-action=\\"ok\\"
          data-mdc-dialog-button-default=\\"\\"
          type=\\"button\\"
        >
          <div
            class=\\"mdc-button__ripple\\"
          />
          <span
            class=\\"mdc-button__label\\"
          >
            OK
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class=\\"mdc-dialog__scrim\\"
  />
</div>"
`;

exports[`AlertDialog component supports simple dialogs 2`] = `
"<div
  class=\\"mdc-dialog\\"
>
  <div
    class=\\"mdc-dialog__container\\"
  >
    <div
      aria-modal=\\"true\\"
      class=\\"mdc-dialog__surface bar\\"
      role=\\"alertdialog\\"
    >
      <div
        class=\\"mdc-dialog__content\\"
      >
        foo
      </div>
      <div
        class=\\"mdc-dialog__actions\\"
      >
        <button
          class=\\"mdc-button mdc-dialog__button\\"
          data-mdc-dialog-action=\\"cancel\\"
          data-mdc-dialog-button-default=\\"\\"
          type=\\"button\\"
        >
          <div
            class=\\"mdc-button__ripple\\"
          />
          <span
            class=\\"mdc-button__label\\"
          >
            Cancel
          </span>
        </button>
        <button
          class=\\"mdc-button mdc-dialog__button\\"
          data-mdc-dialog-action=\\"ok\\"
          type=\\"button\\"
        >
          <div
            class=\\"mdc-button__ripple\\"
          />
          <span
            class=\\"mdc-button__label\\"
          >
            OK
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class=\\"mdc-dialog__scrim\\"
  />
</div>"
`;
